spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/config/PropertySourceBootstrapConfiguration.java

```diff
@@ -154,36 +164,39 @@ private void setLogLevels(ConfigurableApplicationContext applicationContext,
 	}
 
 	private void insertPropertySources(MutablePropertySources propertySources,
-			CompositePropertySource composite) {
+			List<PropertySource<?>> composite) {
 		MutablePropertySources incoming = new MutablePropertySources();
-		incoming.addFirst(composite);
+		List<PropertySource<?>> reversedComposite = new ArrayList(composite);
+		// Reverse the list so that when we call addFirst below we are maintaining the
+		// same order of PropertySournces
+		// Wherever we call addLast we can use the order in the List since the first item
+		// will end up before the rest
+		Collections.reverse(reversedComposite);
+		reversedComposite.stream().forEach(p -> incoming.addFirst(p));
 		PropertySourceBootstrapProperties remoteProperties = new PropertySourceBootstrapProperties();
 		Binder.get(environment(incoming)).bind("spring.cloud.config",
 				Bindable.ofInstance(remoteProperties));
 		if (!remoteProperties.isAllowOverride() || (!remoteProperties.isOverrideNone()
 				&& remoteProperties.isOverrideSystemProperties())) {
-			propertySources.addFirst(composite);
+			reversedComposite.stream().forEach(p -> propertySources.addFirst(p));
```

Comment:
I'm a bit uncomfortable with using Stream APIs in this low level code. Startup and GC performance is quite sensitive to this kind of thing. Can we use a normal iterator?

Reply:
:+1:

Reply:
For sure

Reply:
@dsyer done!

Reply:
:+1: